tests/src/JIT/HardwareIntrinsics/X86/Sse/ReciprocalScalar.cs

```diff
@@ -31,11 +31,12 @@ static unsafe int Main(string[] args)
 
                     if (!floatTable.CheckResult((x, y) => {
                         var expected = 1 / x[0];
+                        var remain = Avx.IsSupported ? (y[1] == x[1]) && (y[2] == x[2]) && (y[3] == x[3]) : true; // VEX-encoding copies remain elements
```

Comment:
Why are these failing now, and not before this change?

Reply:
Ah, because you have it marked as SimpleSIMD, and the 1 arg case goes through `emitIns_R_R`, which doesn't handle the move.

Previously this was going through: `emit->emitIns_SIMD_R_R_R(ins, targetReg, op1Reg, op1Reg, TYP_SIMD16)`, which handled the move for non-VEX.

Reply:
I still say the proper fix is to make `ReciprocalScalar`, `ReciprocalSqrtScalar`, and `SqrtScalar` have the same signature as the other scalar intrinsics (`upper, value`, rather than just `value`).

It makes the handling easier, makes the API surface consistent, and gives the users the option to set the upper bits as they need.

Reply:
They failed with `COMPlus_EnableAVX=0`. 
Originally, you used `emitIns_SIMD_R_R_R(ins, targetReg, op1Reg, op1Reg)` that always copies op1 (src) to the target with SSE encoding, which is unnecessary.

Reply:
>  and gives the users the option to set the upper bits as they need.

But that only makes sense with VEX-encoding.

Reply:
> They failed with COMPlus_EnableAVX=0

Yes, but they weren't failing previously.

> Originally, you used emitIns_SIMD_R_R_R(ins, targetReg, op1Reg, op1Reg) that always copies op1 (src) to the target with SSE encoding, which is unnecessary.

It only copied op1 to the target, if `targetReg != op1Reg`. It also caused the upper bits to be set appropriately.

Reply:
>  It also caused the upper bits to be set appropriately.

So we need the "keeping upper bits" semantics? If yes, we need to describe this behavior in the doc.

Reply:
> But that only makes sense with VEX-encoding.

Why? The behavior of the SSE encoding is:
![image](https://user-images.githubusercontent.com/10487869/35135949-3d8e56bc-fc95-11e7-8c9b-9b6c949def7a.png)

So the upper bits of the target register are unmodified.

The behavior of the Intel native intrinsic is:
![image](https://user-images.githubusercontent.com/10487869/35135977-78d4096a-fc95-11e7-86b4-1bf3adceae39.png)

The behavior of the managed intrinsic should be the same, regardless of AVX or SSE.

Reply:
> So the upper bits of the target register are unmodified.

Your original implementation keeps the upper bits of *source* rather than target.

Reply:
> So we need the "keeping upper bits" semantics

Yes, at a minimum, in order to give the user deterministic behavior. Otherwise, the upper bits are completely random on non-AVX machines.

However, out of all the scalar intrinsics, we currently have 3 that don't let the user explicitly pass in `upper` in the API, and I think rectifying that (to make it consistent) should be the actual fix.

Reply:
> user deterministic behavior.

Sounds good, so we indeed want the semantics of "keeping upper bits of source to target". Right? I will fix.

Reply:
> Your original implementation keeps the upper bits of source rather than target.

My point was, that the SSE behavior, when using the legacy encoding, is to keep the upper bits of the target. With the VEX encoding, the behavior is to select the upper bits from the first source.

On VEX machines, we currently have logic to pass in `value` as both sources.

On non-VEX machines, the previous logic was to mirror that behavior by explicitly copying `value` to the target, and then calling the instruction.

This gives deterministic behavior, regardless of whether you are on an legacy or VEX-enabled machine.

The new behavior makes VEX and non-VEX machines behave differently and results in non-deterministic behavior on legacy machines.